validation/validate.ts

```diff
@@ -1,62 +1,298 @@
+  console.log(result);
+  */
+})();
+
+/**
+ * The return type for runHTMLSnippetComparison.
+ *
+ * Accnames property always present, caseId property
+ * only present if accnames disagree and a testcase
+ * has been saved.
+ */
+interface HTMLSnippetComparisonResult {
+  accnames: {[key: string]: string};
+  caseId?: string;
+}
+
+/**
+ * Compares AccName implementations on a HTML snippet containing a target element
+ * identified by the presence of an 'accnameComparisonTarget' attribute.
+ * @param HTMLSnippet - The HTML snippet containing the target element.
+ * @return An object containing the accessible names computed by 5 different implementations
+ * and, in the case that there was some disagreement between implementations, an ID
+ * for the test case file containing the details of that disagreement.
+ */
+export async function runHTMLSnippetComparison(
+  HTMLSnippet: string
+): Promise<HTMLSnippetComparisonResult> {
+  // Load HTML snippet into Puppeteer browser
   const browser = await puppeteer.launch({
     args: ['--enable-blink-features=AccessibilityObjectModel'],
   });
   const page = await browser.newPage();
-  await page.goto('data:text/html,' + INPUT_HTML);
+  await page.goto('data:text/html,' + HTMLSnippet);
   const client = await page.target().createCDPSession();
 
   await loadAccNameLibraries(page);
+
   const targetNodeRef = await getNodeRefFromSelector(
     '[accnameComparisonTarget]',
     client,
     page
   );
-  if (targetNodeRef) {
-    const accnames = await runComparison(targetNodeRef, page, client);
-    console.log('AccName comparison:\n', accnames);
 
-    const htmlUsedByChrome = await getHTMLUsedByChrome(
-      targetNodeRef,
+  const comparisonResults = await runComparison(targetNodeRef, page, client);
+  await browser.close();
+  if (comparisonResults.disagrees) {
+    const caseId = saveTestcase(comparisonResults, 'output/snippet');
+    return {caseId: caseId, accnames: comparisonResults.accnames};
+  }
+
+  return {accnames: comparisonResults.accnames};
+}
+
+/**
+ * Compares AccName implementations on every Node in the DOM of the web page
+ * at the URL provided.
+ * @param url - The URL for the web page whose Nodes will be used to compare
+ * AccName implementations.
+ * @return The name of the directory containing the results of the URL comparison.
+ */
+export async function runURLComparison(url: string): Promise<string> {
+  const browser = await puppeteer.launch({
+    args: ['--enable-blink-features=AccessibilityObjectModel'],
+  });
+  const page = await browser.newPage();
+  const client = await page.target().createCDPSession();
+
+  const httpResponse = await page.goto(url);
+  if (!httpResponse?.ok()) {
+    throw new Error(`URL '${url}' could not be accessed.'`);
+  }
+
+  await loadAccNameLibraries(page);
+
+  // Trim URL string to get a concise directory name for this comparison.
+  const dirName = url.replace(/^(?:https?:\/\/)?(?:www\.)?/i, '').split('/')[0];
```

Comment:
What if I want to run a test on both `example.com/first` and `example.com/second`? They'll end up in the same directory?